src/UnitTests/Scanning/ScannerTests.cs

```diff
@@ -518,6 +518,79 @@ public void Scanner_Trampoline()
         }
 
         [Test]
+        public void Scanner_NoDecompledProcedure()
+        {
+            Given_Program(Address.Ptr32(0x1000), new byte[0x2000]);
+            program.User.Procedures.Add(
+                Address.Ptr32(0x2000),
+                new Procedure_v1()
+                {
+                    Name = "ndProc",
+                    CSignature = "int ndProc(double dVal)",
+                    Decompile = false,
+                }
+            );
+
+            var sc = CreateScanner(program);
+            var proc = sc.ScanProcedure(Address.Ptr32(0x2000), "fn000020", arch.CreateProcessorState());
+            Assert.AreEqual("ndProc", proc.Name);
+            Assert.AreEqual("int32", proc.Signature.ReturnValue.DataType.ToString());
+            Assert.AreEqual("eax", proc.Signature.ReturnValue.Name);
+            Assert.AreEqual(1, proc.Signature.Parameters.Length);
+            Assert.AreEqual("real64", proc.Signature.Parameters[0].DataType.ToString());
+            Assert.AreEqual("dVal", proc.Signature.Parameters[0].Name);
+        }
+
+        [Test]
+        public void Scanner_EnqueueUserProcedure()
+        {
+            Given_Program(Address.Ptr32(0x1000), new byte[0x2000]);
+
+            var sc = CreateScanner(program);
+            sc.EnqueueUserProcedure(new Procedure_v1
+            {
+                Address = "0x1010",
+                Name = "proc1",
+                Decompile = false
+            });
+            sc.EnqueueUserProcedure(new Procedure_v1
+            {
+                Address = "0x1020",
+                Name = "proc2",
+            });
+            sc.ScanImage();
+            Assert.AreEqual(1, program.Procedures.Count);
+            Assert.AreEqual("proc2", program.Procedures.Values[0].Name);
+        }
+
+        [Test]
+        public void Scanner_NoDecompiledEntryPoint()
+        {
+            Given_Program(Address.Ptr32(0x12314), new byte[20]);
+            program.User.Procedures.Add(
+                Address.Ptr32(0x12314),
+                new Procedure_v1()
+                {
+                    Decompile = false,
+                }
+            );
+
+            var sc = CreateScanner(program);
+            sc.EnqueueEntryPoint(
+                new EntryPoint(
+                    Address.Ptr32(0x12314),
+                    program.Architecture.CreateProcessorState()));
+            sc.EnqueueEntryPoint(
```

Comment:
Are the two EntryPoints intentional? If so, add a comment explaining why there are two entry points with same address, otherwise remove the duplicate.